sampler.go

```diff
@@ -541,16 +554,16 @@ func (s *RemotelyControlledSampler) updateAdaptiveSampler(strategies *sampling.P
 
 // NB: this function should only be called while holding a Write lock
 func (s *RemotelyControlledSampler) updateRateLimitingOrProbabilisticSampler(res *sampling.SamplingStrategyResponse) error {
-	var newSampler Sampler
 	if probabilistic := res.GetProbabilisticSampling(); probabilistic != nil {
-		newSampler = newProbabilisticSampler(probabilistic.SamplingRate)
+		s.sampler = newProbabilisticSampler(probabilistic.SamplingRate)
 	} else if rateLimiting := res.GetRateLimitingSampling(); rateLimiting != nil {
-		newSampler = NewRateLimitingSampler(float64(rateLimiting.MaxTracesPerSecond))
+		if rateLimitingSampler, ok := s.sampler.(*rateLimitingSampler); ok {
+			rateLimitingSampler.update(float64(rateLimiting.MaxTracesPerSecond))
```

Comment:
note that the update() method has no short-circuit for unchanged rate. Maybe it shoud?

Reply:
donito